File: MauiFrontend/MauiFrontend/Services/PrintService.cs
using MauiFrontend.Models;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Net.Http.Json;
using System.Text;
using System.Threading.Tasks;

namespace MauiFrontend.Services
{
    public class PrintService
    {
        private readonly HttpClient _http;
        private const string BaseUrl = "http://10.0.2.2:5128/api/printer";

        public PrintService()
        {
            _http = new HttpClient();
        }

        public async Task<List<Printer>> GetPrintersAsync()
        {
            try
            {
                var printers = await _http.GetFromJsonAsync<List<Printer>>(BaseUrl);
                return printers;
            }
            catch (Exception exception)
            {
                Debug.WriteLine(exception.Message);
                return new();
            }
        }

        public async Task SubmitPrintJobAsync(int printerId)
        {
            var job = new PrintJob
            {
                PrinterId = printerId,
                DocumentId = printerId
            };

            await _http.PostAsJsonAsync($"{BaseUrl}/print", job);
        }
    }
}

File: FFrontend/stage_uitvoeringen/lib/services/print_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../models/printer.dart';
import '../models/print_job.dart';

class PrintService {
  final String baseUrl = 'http://10.0.2.2:5128/api/printer';
  final http.Client client = http.Client();

  Future<List<Printer>> getPrintersAsync() async {
    try {
      final response = await client.get(Uri.parse(baseUrl));

      if (response.statusCode == 200) {
        final List<dynamic> data = jsonDecode(response.body);
        final printers = data.map((item) => Printer.fromJson(item)).toList();

        return printers;
      } else {
        print('Error fetching printers: ${response.statusCode}');
        return [];
      }
    } catch (e) {
      print('Exception in getPrintersAsync: $e');
      return [];
    }
  }

  Future<void> submitPrintJobAsync(int printerId) async {
    final job = PrintJob(
      printerId: printerId,
      documentId: printerId, // mimicking the same as in C#
    );

    try {
      final response = await client.post(
        Uri.parse('$baseUrl/print'),
        headers: {'Content-Type': 'application/json'},
        body: jsonEncode(job.toJson()),
      );

      if (response.statusCode >= 200 && response.statusCode < 300) {
        print('Print job submitted successfully');
      } else {
        print('Failed to submit print job: ${response.statusCode}');
      }
    } catch (e) {
      print('Exception in submitPrintJobAsync: $e');
    }
  }
}
